redfish-core/lib/managers.hpp

```diff
@@ -1962,6 +1962,55 @@ inline void
     });
 }
 
+inline void getBMCState(const std::shared_ptr<bmcweb::AsyncResp>& aResp)
+{
+    aResp->res.jsonValue["PowerState"] = "On";
+    sdbusplus::asio::getProperty<std::string>(
+        *crow::connections::systemBus, "xyz.openbmc_project.State.BMC",
+        "/xyz/openbmc_project/state/bmc0", "xyz.openbmc_project.State.BMC",
+        "CurrentBMCState",
+        [aResp](const boost::system::error_code ec,
+                const std::string& bmcState) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error reading CurrentBmcState");
+            aResp->res.jsonValue["Status"]["State"] = "Enabled";
+            aResp->res.jsonValue["Status"]["Health"] = "OK";
+            return;
+        }
+
+        if (bmcState == "xyz.openbmc_project.State.BMC.BMCState.Ready")
+        {
+            aResp->res.jsonValue["Status"]["State"] = "Enabled";
+            aResp->res.jsonValue["Status"]["Health"] = "OK";
+        }
+        else if (bmcState == "xyz.openbmc_project.State.BMC.BMCState."
+                             "Quiesced")
+        {
+            aResp->res.jsonValue["Status"]["State"] = "Quiesced";
+            aResp->res.jsonValue["Status"]["Health"] = "Critical";
+        }
+        else if (bmcState == "xyz.openbmc_project.State.BMC.BMCState."
+                             "NotReady")
+        {
+            aResp->res.jsonValue["Status"]["State"] = "Starting";
+            aResp->res.jsonValue["Status"]["Health"] = "OK";
+        }
+        else if (bmcState == "xyz.openbmc_project.State.BMC.BMCState."
+                             "UpdateInProgress")
+        {
+            aResp->res.jsonValue["Status"]["State"] = "Updating";
+            aResp->res.jsonValue["Status"]["Health"] = "OK";
+        }
+        else
+        {
+            BMCWEB_LOG_DEBUG("Unsupported D-Bus CurrentBMCState: {}", bmcState);
```

Comment:
Not sure why you need this, but 🤷 
You are sweeping this error under the rug. Typically we would do an internal error here but since this is what we had before 🤷 I am okay with it

Reply:
I think it was to stick with the previous bmcweb default (hard code to this to start with). I'm pretty in-tune with any new bmc states so lets just leave as-is.

Reply:
Ack